### PR TITLE
android: update build scripts

### PR DESCRIPTION
**Pull request by alalek**
Force using of the same CMake & NDK version during the build.

```
force_builders_only=Android armeabi-v7a,Android pack
```

### CODE REVIEW

File: cmake/android/android_gradle_projects.cmake [hunk 1]
```diff
@@ -80,6 +82,15 @@ foreach(fname ${GRADLE_WRAPPER_FILES})
   install(FILES "${OpenCV_SOURCE_DIR}/platforms/android/gradle-wrapper/${fname}" DESTINATION "${ANDROID_INSTALL_SAMPLES_DIR}/${__dir}" COMPONENT samples ${__permissions})
 endforeach()
 
+# force reusing of the same CMake version
+if(NOT OPENCV_SKIP_ANDROID_FORCE_CMAKE)
+  if(NOT DEFINED _CMAKE_INSTALL_DIR)
+    get_filename_component(_CMAKE_INSTALL_DIR "${CMAKE_ROOT}" PATH)
+    get_filename_component(_CMAKE_INSTALL_DIR "${_CMAKE_INSTALL_DIR}" PATH)
```
Comment (asmorkalov) [inline cmake/android/android_gradle_projects.cmake:89]: It overrides variable value in the previous line.

File: samples/android/build.gradle.in [hunk 2]
```diff
@@ -90,5 +91,14 @@ gradle.afterProject { project ->
             }
         }
 
+        // Android Gradle Plugin (AGP) 3.5+ is required
+        // https://github.com/android/ndk-samples/wiki/Configure-NDK-Path
+        def isNdkVersionSupported = project.android.metaClass.getProperties().find { it.name == 'ndkVersion' } != null
+        if ((false || opencv_strict_build_configuration) && isNdkVersionSupported) {
```
Comment (asmorkalov) [inline samples/android/build.gradle.in:97]: why do you need false here?

File: cmake/android/android_gradle_projects.cmake (hunk 1 above)
Comment (alalek) [reply cmake/android/android_gradle_projects.cmake:89]: Yes. It is intended as we need `../../` location.

File: samples/android/build.gradle.in (hunk 2 above)
Comment (alalek) [reply samples/android/build.gradle.in:97]: This option may be useful on the user side (this is a part of OpenCV Android SDK), but it is not enabled by default (to keep Android "defaults" - which depends on versions of user's Android environment).

To manually enable this option just switch `false->true`.

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**